Libplanet.Net/Transports/NetMQTransport.cs

```diff
@@ -808,98 +804,64 @@ private async Task ProcessRequest(MessageRequest req, CancellationToken cancella
-                        }
-                        catch (DifferentAppProtocolVersionException dapve)
-                        {
-                            const string dbgMsg =
-                                "Received reply {Reply} from {Peer} to request {Message} " +
-                                "{RequestId} has an invalid APV.";
-                            _logger.Debug(
-                                dapve,
-                                dbgMsg,
-                                reply,
-                                reply.Remote,
-                                message,
-                                req.Id);
-                            throw;
-                        }
-
-                        result.Add(reply);
+                            message,
+                            req.Id);
+                        throw;
                     }
-                    catch (OperationCanceledException oce)
+                    catch (DifferentAppProtocolVersionException dapve)
                     {
-                        if (timerCts.IsCancellationRequested)
-                        {
-                            if (req.ReturnWhenTimeout)
-                            {
-                                break;
-                            }
-
-                            throw new TimeoutException(
-                                $"The operation was canceled due to timeout {req.Timeout}.",
-                                oce);
-                        }
-
+                        const string dbgMsg =
+                            "Received reply {Reply} from {Peer} to request {Message} " +
+                            "{RequestId} has an invalid APV.";
+                        _logger.Debug(
+                            dapve,
+                            dbgMsg,
+                            reply,
+                            reply.Remote,
+                            message,
+                            req.Id);
                         throw;
                     }
-                }
 
-                tcs.TrySetResult(result);
+                    await channel.Writer.WriteAsync(reply, cancellationToken);
+                }
             }
-            catch (Exception e) when (
-                e is SendMessageFailException ||
-                e is InvalidMessageSignatureException ||
-                e is InvalidMessageTimestampException ||
-                e is DifferentAppProtocolVersionException ||
-                e is TimeoutException)
+            catch (Exception e)
             {
-                tcs.TrySetException(e);
-            }
-            catch (Exception ae)
-            {
-                var se = new SendMessageFailException(
-                    $"Unexpected exception occurred during {nameof(ProcessRequest)}().",
-                    req.Peer,
-                    ae
-                );
-                tcs.TrySetException(se);
+                channel.Writer.Complete(e);
```

Comment:
Perhaps may not be strictly necessary, but maybe we should write `Complete()` on successful operation of `for` loop also? 😶

Reply:
Like https://github.com/planetarium/libplanet/pull/2522/files#r1018739172 ?

Reply:
Added https://github.com/planetarium/libplanet/pull/2522/commits/5a41085db23b04e8ead8abb0b3d66259c38e4f28.